### DISCUSSION TITLE
Setup and networking with Proxmox under WSL Debian

### DISCUSSION THREAD

**Discussion by SemanticBeeng** [ehlesp/smallab-k8s-pve-guide#13] [General]
Trying to setup a homelab using proxmox and k8s and started using WSL2 Debian for convenience.
Having issues with networking and trying different things.

The knowledge you put together is very valuable, thanks!
But does not get me "there" just yet. 

Also finding this https://github.com/gaudat/gauup/blob/master/homelab/pve-wsl2.md useful and trying to combine with
https://github.com/ehlesp/smallab-k8s-pve-guide/blob/main/G017%20-%20Virtual%20Networking%20~%20Network%20configuration.md 

Also this setup is great https://mayfly277.github.io/posts/GOAD-on-proxmox-part1-install/ .

Maybe you would be kind to determine if combining these resources improves this knowledge.

**Comment by ehlesp** [comment]
I'd say that by using bits and pieces from the three documents you've linked, yes you might eventually get where you want, although you don't really tell what you want to achieve. Just be ready for quite a bit of trial and error, because you'll have to combine guides written for three different scenarios and different goals. On the other hand, I'd use VirtualBox for setting up the ProxmoxVE virtual machine, I think it would be easier. Also, ensure you have a good machine with plenty of RAM to run that ProxmoxVE VM if you plan to create a K8s cluster inside of it. And be mindful of enabling for your ProxmoxVE VM access to the virtualization instructions your homelab host's CPU should have. Otherwise, ProxmoxVE won't run properly! With VirtualBox you can do it with some shell command, but I don't know if you can do it with WSL.
